tensorflow/core/util/cuda_kernel_helper.h

```diff
@@ -432,6 +433,43 @@ CUDA_ATOMIC_WRAPPER(Add, double) {
   return __longlong_as_double(old);
 }
 
+// Custom implementation of atomicAdd for std::complex<float>.
+// This implementation performs to atomic additions on the components.
+CUDA_ATOMIC_WRAPPER(Add, std::complex<float>) {
+#if defined(__CUDA_ARCH__)
+#if __CUDA_ARCH__ >= 350
+  float2* addr_as_float2 = reinterpret_cast<float2*>(address);
+  float2* val_as_float2 = reinterpret_cast<float2*>(&val);
+  CudaAtomicAdd(&(addr_as_float2->x), val_as_float2->x);
+  CudaAtomicAdd(&(addr_as_float2->y), val_as_float2->y);
+#else
+  static_assert(false,
+                "Unable to compile CudaAtomicAdd for complex64 because "
+                "architectures < sm35 are not supported");
+#endif
+#endif
+  return *address;
+}
+
+// Custom implementation of atomicAdd for std::complex<double>.
+// This implementation performs to atomic additions on the components
+// using the double atomic wrapper above.
+CUDA_ATOMIC_WRAPPER(Add, complex128) {
+#if defined(__CUDA_ARCH__)
+#if __CUDA_ARCH__ >= 350
+  double2* addr_as_double2 = reinterpret_cast<double2*>(address);
+  double2* val_as_double2 = reinterpret_cast<double2*>(&val);
+  CudaAtomicAdd(&(addr_as_double2->x), val_as_double2->x);
+  CudaAtomicAdd(&(addr_as_double2->y), val_as_double2->y);
+#else
+  static_assert(false,
```

Comment:
@ebrevdo Jenkins GPU tests seem to be failing here. Any ideas on how to fix this?

Reply:
Why do we set cuda min compute capabilities to 30?  I thought we supported only 35 and up?  The error is:

./tensorflow/core/util/cuda_kernel_helper.h(465): error: static assertion failed with "Unable to compile CudaAtomicAdd for complex128 because architectures < sm35 are not supported

Reply:
I'm not too familiar with the testing setup @gunan might know more.